cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -101,26 +117,66 @@ int MetricsCollector::RecordMetrics(
+    Aws::CloudWatchMetrics::Utils::MetricObject metric_object{metric_msg->metric_name,
                                                               metric_msg->value,
                                                               metric_msg->unit,
-                                                              GetMetricDataEpochMillis(*metric_msg),
+                                                              GetMetricDataEpochMillis(metric_msg->time_stamp),
                                                               dimensions,
                                                               this->storage_resolution_.load()};
     bool batched = metric_service_->batchData(metric_object);
 
     if (!batched) {
-      AWS_LOGSTREAM_ERROR(__func__, "Failed to record metric");
+      AWS_LOGSTREAM_ERROR(__func__, "Failed to record ROS monitoring message");
+    } else {
+      ++batched_count;
     }
-
-    batched_count++;
   }
   return batched_count;
 }
 
-int64_t MetricsCollector::GetMetricDataEpochMillis(const ros_monitoring_msgs::msg::MetricData & metric_msg)
+int MetricsCollector::RecordMetrics(
+  metrics_statistics_msgs::msg::MetricsMessage::UniquePtr msg)
+{
+  AWS_LOGSTREAM_DEBUG(__func__, "Received metric with " << msg->statistics.size() << " types of statistics");
+  AWS_LOGSTREAM_DEBUG(__func__, "Recording metric with name=[" << msg->measurement_source_name << "]");
+
+  // create a MetricObject with message parameters to batch
+  Aws::CloudWatchMetrics::Utils::MetricObject metric_object;
+  metric_object.metric_name = std::move(msg->measurement_source_name);
+  metric_object.unit = std::move(msg->unit);
+  metric_object.timestamp = GetMetricDataEpochMillis(msg->window_start);
+  metric_object.value = std::numeric_limits<double>::quiet_NaN();
+  metric_object.storage_resolution = this->storage_resolution_.load();
+
+  metric_object.dimensions["metric_type"] = std::move(msg->metrics_source);
+  metric_object.dimensions["duration"] = GetMetricDataEpochMillis(msg->window_stop) - metric_object.timestamp;
+
+  for (const auto & statistic : msg->statistics) {
+    if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_AVERAGE) {
+      metric_object.value = statistic.data;
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_MAXIMUM) {
+      metric_object.statistic_values[StatisticValuesType::MAXIMUM] = statistic.data;
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_MINIMUM) {
+      metric_object.statistic_values[StatisticValuesType::MINIMUM] = statistic.data;
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_SAMPLE_COUNT) {
+      metric_object.statistic_values[StatisticValuesType::SAMPLE_COUNT] = statistic.data;
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_STDDEV) {
+      metric_object.dimensions["stddev"] = std::to_string(statistic.data);
+    }
```

Comment:
This conversion does not make sense to me for a number of reasons, listing out a few: 

1. Dimension values are typically discrete and therefore queryable. For. eg. server=prod or region=us-west-2. By have standard-deviation values as dimension values you are making the metric next to impossible to query because we wont know the stddev values upfront. 

2. Lets look at how this metric datum will translate to if the same thing were done using cli 
```
{
  "MetricName":"cpu",
  "Dimensions":[
    {
      "Name":"stdev",
      "Value":"0.5"
    }
  ],
  "StatisticValues":{
    "SampleCount":10,
    "Sum":1000,
    "Minimum":20,
    "Maximum":100
  }
}
```
This makes no sense as If now want to find out the avg MAX cpu values, I will need to know all the stddev values as I cannot use [cloudwatch metric math](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/using-metric-math.html) to retrieve statistics without specifying dimension values. 

I understand that cloudwatch does not provide a standdev option in [statistic value](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Statistic).

A better alternative is to record the stddev as a regular metric with no unit type.
You can create a new metric named `stddev` name spaced under ` metric_object.metric_name` and have `None` unit type and the stddev value as `Value`.

Reply:
Addressed in latest update to pull request.

Reply:
Uh oh:
```
Encountered AWSError 'InvalidParameterCombination': The parameters MetricData.member.1.Value and MetricData.member.1.StatisticValues are mutually exclusive and you have specified both.
```

So we can't pack duration into a `MetricObject` that already has average, min, max, count, etc. the way we are trying to do it now.